x/connectivity/errors_windows.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// go:build windows
+
+package connectivity
+
+import (
+	"fmt"
+	"golang.org/x/sys/windows"
+	"syscall"
+)
+
+func errnoName(errno syscall.Errno) string {
+	// List from https://cs.opensource.google/go/x/sys/+/master:windows/zerrors_windows.go,
```

Comment:
Just FYI, here is the official list of Windows Socket error codes: https://learn.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2